docs(tutorials): elk stack (#2230)
scaleway/docs-content: tutorials/collecting-visualizing-logs-elastic-stack/index.mdx

  - instances
  - elastic-metal
dates:
  validation: 2023-10-03
  posted: 2015-06-10
---

> Elastic Stack, formerly known as ELK Stack is a bundle of three open-source software projects maintained by [Elastic](https://www.elastic.co/). The stack comprises Elasticsearch, Logstash, Kibana, and Beats. 

Reviewer: ```suggestion
Elastic Stack, formerly known as ELK Stack, is a bundle of three open-source software projects maintained by [Elastic](https://www.elastic.co/). The stack comprises Elasticsearch, Logstash, Kibana, and Beats. 
```

   apt update
   ```

4. Install Elasticsearch:
   ```bash
   apt install elasticsearch
   ```

5. Start and enable the Elasticsearch service:
   ```bash
   sudo systemctl start elasticsearch
   sudo systemctl enable elasticsearch
   ```

## Install and configure Logstash

1. Using the same repository added for Elasticsearch, you can simply install Logstash:
   ```bash
   apt install logstash
   ```

2. Once installed, you can create and modify configuration files for Logstash to set up your data pipelines. These are typically found in `/etc/logstash/conf.d/`.

3. Start and enable the Logstash service:
   ```bash
   sudo systemctl start logstash
   sudo systemctl enable logstash
   ```

## Install and configure Kibana

1. Install Kibana using the repository:
   ```bash
   apt install kibana
   ```

2. Start and enable the Kibana service:
   ```bash
   sudo systemctl start kibana
   sudo systemctl enable kibana
   ```

3. By default, Kibana is accessible on `http://localhost:5601`. If you need to access it from a remote machine, edit the Kibana configuration file `/etc/kibana/kibana.yml` and set the server host:
   ```
   server.host: "0.0.0.0"
   ```

## Secure the Elastic stack

> It it important to secure your ELK Stack, especially if it is exposed to the public internet. Follwoing are some additional resources to complete your setup:

Reviewer 2: ```suggestion
It is important to secure your ELK Stack, especially if it is exposed to the public internet. You can complete your setup using the following additional resources:
```
